src/Exporter.php

```diff
@@ -337,12 +344,23 @@ private function exportObject(mixed $value, RecursionContext $processed, string
 
         $processed->add($value);
 
-        $values = '';
-        $array  = $this->toArray($value);
+        if ($this->objectExporter !== null && $this->objectExporter->handles($value)) {
+            $buffer = $this->objectExporter->export($value, $this);
```

Comment:
We'd need the indentation here as well, both for printing and for passing back to `export`. My thinking here is that if I have an entity (whatever the provider), my code would:
1. Get properties of that entity (excluding any infrastructure-supporting instance properties)
2. For each property print the name
3. If it is a collection
3.1. Iterate over the collection
3.2. Go recursively for each entry
4. If it is not a collection
5. Call the `export` method with the indentation increased accordingly. _That_ way I can defer to the default functionality.

Reply:
Good catch, implemented.